wsd/COOLWSD.cpp

```diff
@@ -1003,6 +1010,254 @@ COOLWSD::~COOLWSD()
+            if (!conf.has(path))
+            {
+                break;
+            }
+            newAppConfig.insert(std::make_pair(path, ""));
+            newAppConfig.insert(std::make_pair(path + "[@allow]", "false"));
+        }
+    }
+
+    void fetchLockedHostPatterns(std::map<std::string, std::string>& newAppConfig, Poco::JSON::Object::Ptr remoteJson)
+    {
+        if (!conf.getBool("feature_lock.locked_hosts[@allow]", false))
+        {
+            LOG_INF("locked_hosts feature is disabled from coolwsd.xml");
+            return;
+        }
+
+        Poco::JSON::Array::Ptr lockedHostPatterns =
+            remoteJson->getObject("feature_locking")->getObject("locked_hosts")->getArray("hosts");
+
+        if (lockedHostPatterns->size() == 0)
+        {
+            LOG_WRN("Not overwriting any locked wopi host pattern because JSON contains empty array");
+            return;
+        }
+
+        std::size_t i;
+        for (i=0 ; i < lockedHostPatterns->size(); i++)
+        {
+            std::string host;
+
+            Poco::JSON::Object::Ptr subObject = lockedHostPatterns->getObject(i);
+            JsonUtil::findJSONValue(subObject, "host", host);
+            Poco::Dynamic::Var readOnly = subObject->get("read_only");
+            Poco::Dynamic::Var disabledCommands = subObject->get("disabled_commands");
+
+            booleanToString(readOnly);
+
+            booleanToString(disabledCommands);
+
+            const std::string path = "feature_lock.locked_hosts.host[" + std::to_string(i) + "]";
+            newAppConfig.insert(std::make_pair(path, host));
+            newAppConfig.insert(std::make_pair(path + "[@read_only]", readOnly));
+            newAppConfig.insert(std::make_pair(path + "[@disabled_commands]", disabledCommands));
+        }
+
+        //if number of locked wopi host patterns defined in coolwsd.xml are greater than number of host
+        //fetched from json, overwrite the remaining host from config file to empty strings and
+        //set read_only and disabled_commands to false
+        for (; ; ++i)
+        {
+            const std::string path = "feature_lock.locked_hosts.host[" + std::to_string(i) + "]";
+            if (!conf.has(path))
+            {
+                break;
+            }
+            newAppConfig.insert(std::make_pair(path, ""));
+            newAppConfig.insert(std::make_pair(path + "[@read_only]", "false"));
+            newAppConfig.insert(std::make_pair(path + "[@disabled_commands]", "false"));
+        }
+    }
+
+    //sets property to false if it is missing from JSON
+    //and convert the property to string
+    void booleanToString(Poco::Dynamic::Var& booleanFlag)
+    {
+        if (!booleanFlag.isEmpty())
+        {
+            if (booleanFlag.isBoolean())
+            {
+                booleanFlag.toString();
+            }
+        }
+        else
+        {
+            booleanFlag = "false";
+        }
+    }
+
+    enum ConfigPriority
```

Comment:
Wouldn't it be better to use a more modern scoped enum (enum class) here? And then the PRIO_ prefix of the PRIO_JSON enumerator would be unnecessary as already the ConfigPriority identifier contains "Priority".

Reply:
Or wait, as the PRIO_JSON value is passed to a function that expects an int, why use an enum at all, why not just constexpr int PRIO_JSON = -200;?

Reply:
ok